core/converter/camel-case-converter.js

```diff
@@ -0,0 +1,29 @@
+/**
+ * @module montage/core/converter/camel-case-converter
+ * @requires montage/core/converter/converter
+ */
+var Converter = require("./converter").Converter,
+    camelCase = require('lodash/fp/camelCase'),
+    singleton;
+
+/**
+ * Converts string to camel case.
+ *
+ * @class CamelCaseConverter
+ * @extends Converter
+ */
+var CamelCaseConverter = exports.CamelCaseConverter = Converter.specialize({
+    convert: {
+        value: function (v) {
+            return camelCase(v);
+        }
+    }
+});
+
+Object.defineProperty(exports, 'defaultCamelCaseConverter', {
```

Comment:
I'm not sure how that defaultCamelCaseConverter pattern works. From what I understand, if someone does:
var CamelCaseConverter = require("montage/core/converter/camel-case-converter").CamelCaseConverter,
camelCaseConverterA = new CamelCaseConverter(),
camelCaseConverterB = new CamelCaseConverter();

then camelCaseConverterA !== camelCaseConverterB. If it's going to be a singleton I think it has to be by default, meaning the constructor needs to return the singleton, always. If one has to use 
var CamelCaseConverter = require("montage/core/converter/camel-case-converter"). defaultCamelCaseConverter

for that, then it wouldn't work as seamlessly in serialization and would likely be ignored

Reply:
The problem is the following, at the moment we declare a converter within the serialization as such:
```
"converter": {
    "prototype": "montage/core/converter"
}
``` 
Which create a new instance of a converter, but we all agree recently (the work on .mjson files) that the key `prototype` should be used for new instance and `object` for object, singleton...
Plus, it could be strange for a developer to always get a singleton after calling the `new` operator.

I believe something should probably be done at the deserialization time.

Reply:
In software engineering, the singleton pattern is a software design pattern that:
- Ensure that only one instance of a class is created.
- Provide a global point of access to the object.

Since these objects are stateless, which is the root idea behind making them singletons, there is no reason to burden the developer with that knowledge when we have a way, creating a singleton object and returning it in the constructor, only when asked. To the outside the developer is getting an instance of the class he wants and that's all he needs to know

Reply:
@corentin 
I don't like that the constructors for the converters return the singleton. If I call `new` on a Montage object I want a new instance, not a singleton. A singleton should be exposed in a constructor property, or for serialization exposed as a separate object in the exports that we could reference with something like
```json
{
  "object": "core/converter/camel-case-converter[instance]"
}
```

Reply:
@marchant I don't like it either I think It's a little bit confusing